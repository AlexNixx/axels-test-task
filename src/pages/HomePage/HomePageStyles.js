import styled from 'styled-components';
import { Col, Container, Row } from 'react-bootstrap';

export const StyledContainer = styled(Container)`
    padding: 1rem;
`;

export const StyledRow = styled(Row)`
    row-gap: 2rem;
`;

export const StyledCol = styled(Col)`
    display: flex;
    justify-content: center;
`;
