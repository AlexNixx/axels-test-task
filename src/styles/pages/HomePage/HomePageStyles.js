import styled from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';

export const StyledContainer = styled(Container)`
    padding: 2rem 1rem;
    overflow: hidden;
`;

export const StyledRow = styled(Row)`
    row-gap: 2rem;
`;

export const StyledCol = styled(Col)`
    display: flex;
    justify-content: center;
`;
