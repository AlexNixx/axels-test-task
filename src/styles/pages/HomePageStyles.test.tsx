import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { StyledCol, StyledContainer, StyledRow } from './HomePageStyles';

describe('Test HomePageStyles', () => {
    test('Check StyledContainer styles', () => {
        const tree = renderer.create(<StyledContainer />).toJSON();
        expect(tree).toHaveStyleRule('padding', '2rem 1rem');
        expect(tree).toHaveStyleRule('overflow', 'hidden');
        expect(tree).toMatchSnapshot();
    });

    test('Check StyledRow styles', () => {
        const tree = renderer.create(<StyledRow />).toJSON();
        expect(tree).toHaveStyleRule('row-gap', '2rem');
        expect(tree).toMatchSnapshot();
    });

    test('Check StyledCol styles', () => {
        const tree = renderer.create(<StyledCol />).toJSON();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toMatchSnapshot();
    });
});
