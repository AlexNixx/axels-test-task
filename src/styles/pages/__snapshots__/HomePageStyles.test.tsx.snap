// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomePageStyles Check StyledCol styles 1`] = `
.c0 {
  display: flex;
  justify-content: center;
}

<div
  className="c0 col"
/>
`;

exports[`Test HomePageStyles Check StyledContainer styles 1`] = `
.c0 {
  padding: 2rem 1rem;
  overflow: hidden;
}

<div
  className="c0 container"
/>
`;

exports[`Test HomePageStyles Check StyledRow styles 1`] = `
.c0 {
  row-gap: 2rem;
}

<div
  className="c0 row"
/>
`;
